### DISCUSSION TITLE
I want to use this to replace Socket.io can get any hint?

### DISCUSSION THREAD

**Discussion by vikastechextensor** [serverpod/serverpod#2066] [Q&A]
For my flutter mobile app, I use node js API server for post/delete calls + Socket io (instead of GET API calls) to keep my data live and in sync.  Socket events are getting passed dynamic both ways. Now I want to use serverpod and so replacing api server and socket.io into serverpod. Can i send and recieve model-based/dynamic data just like socket.io? how? I cant figure out serverpod chat thing can help me or I should try diffrerent way? any dude here to help me?

**Comment by SandPod** [comment]
Hey @vikastechextensor 👋 

Happy to see you testing out Serverpod!

Serverpod comes with a bi-directional Stream interface based on web sockets.

Streams in Serverpod should provide you with a similar experience as Socket.io with the added benefit of our model serialization/deserialization.

It is simple to pass model objects from the client to the server through the `sendStreamMessage` message that is implemented on both the client and the server endpoint.

Our documentation for streams is available here: https://docs.serverpod.dev/concepts/streams
Make sure to read the full page and reply here if you have any additional questions.
